src/middlewared/middlewared/plugins/vm/vms.py

```diff
@@ -175,7 +175,7 @@ async def __common_validation(self, verrors, schema_name, data, old=None):
         if not data.get('uuid'):
             data['uuid'] = str(uuid.uuid4())
 
-        if data['min_memory'] and data['min_memory'] > data['memory']:
+        if data.get('min_memory') is not None and data['min_memory'] > data['memory']:
```

Comment:
Why don't we just make it `default=None`? You can even add `strict=True` to the `Dict` declaration and middleware will crash on startup if any of the dict members is not required and does not have any default value. It's better to have all default values explicitly specified in one place.